companion/lib/Surface/USB/Loupedeck.ts

```diff
@@ -312,8 +356,34 @@ export class SurfaceUSBLoupedeck extends EventEmitter<SurfacePanelEvents> implem
 					})
 			} else if (control.feedbackType === 'lcd') {
 				this.#writeQueue.queue(control.id, render)
+			} 
+		} else if (control.type === 'lcd-segment') { // Update the slider display on render call to change color
+				const styleBg =
+					typeof render.style === 'object' && render.style !== null && 'bgcolor' in render.style
+						? (render.style as any).bgcolor
+						: undefined
+
+				if (control.id == "left") {
+					this.DisplayColors.LeftColor = (({ r, g, b }) => ({ red: r, green: g, blue: b }))(colorToRgb(styleBg));
+				} else if (control.id == "right") {
+					this.DisplayColors.RightColor = (({ r, g, b }) => ({ red: r, green: g, blue: b }))(colorToRgb(styleBg));
+				}
```

Comment:
I think this is quite hard to understand what is going on here.
I find this to be simpler and more readable:

```suggestion
				if (control.id == "left") {
					const col = colorToRgb(styleBg)
					this.DisplayColors.LeftColor = { red: col.r, green: col.g, blue: col.b }
				} else if (control.id == "right") {
					const col = colorToRgb(styleBg)
					this.DisplayColors.RightColor = { red: col.r, green: col.g, blue: col.b }
				}
```